### PR TITLE
fix: update StartDeploymentSession to return default stack name

### PR DESCRIPTION
**Pull request by philasmar**
*Issue #, if available:*
IDE-5945

*Description of changes:*
fix: update StartDeploymentSession to return default stack name

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs [hunk 1]
```diff
@@ -81,7 +81,19 @@ await _projectParserUtility.Parse(input.ProjectPath)
 
             _stateServer.Save(output.SessionId, state);
 
-            output.DefaultDeploymentName = _cloudApplicationNameGenerator.GenerateValidName(state.ProjectDefinition, new List<CloudApplication>());
+            var deployedApplicationQueryer = serviceProvider.GetRequiredService<IDeployedApplicationQueryer>();
+            var session = CreateOrchestratorSession(state);
+            var orchestrator = CreateOrchestrator(state);
+
+            // Determine what recommendations are possible for the project.
+            var recommendations = await orchestrator.GenerateDeploymentRecommendations();
```
Comment (normj) [inline src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs:89]: Should we save `recommendations` to `state.NewRecommendations` and `existingApplications` to `state.ExistingDeployments` so we don't have to recomputed those? That would mean those controller getter methods should look to see if the state already has a value before doing all the work again.

**Review by normj — CHANGES_REQUESTED** [review]
There should be a test to go along with this change.

File: src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs (hunk 1 above)
Comment (philasmar) [reply src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs:89]: Fixed

**Review by normj — APPROVED** [review]

File: src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs [hunk 2]
```diff
@@ -116,7 +130,7 @@ public async Task<IActionResult> GetRecommendations(string sessionId)
 
             var output = new GetRecommendationsOutput();
 
-            state.NewRecommendations = await orchestrator.GenerateDeploymentRecommendations();
+            state.NewRecommendations ??= await orchestrator.GenerateDeploymentRecommendations();
```
Comment (ganeshnj) [inline src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs:133]: above `NewRecommendations` is replaced with newly available recommendations but here only if it is null. A comment/description would help why it is done like this.

**Review by ganeshnj — CHANGES_REQUESTED** [review]

**Pull request merged**